dask-gateway-k8s-operator/pkg/apis/gateway/v1alpha1/daskcluster_types.go

```diff
@@ -0,0 +1,55 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+type Scheduler struct {
+	Template corev1.PodTemplate `json:"template"`
+}
+
+type Worker struct {
+	Template corev1.PodTemplate `json:"template"`
+	Replicas int32              `json:"replicas,omitempty"`
+}
+
+// DaskClusterSpec defines the desired state of DaskCluster
+type DaskClusterSpec struct {
+	Scheduler Scheduler         `json:"scheduler"`
+	Worker    Worker            `json:"worker"`
+	Info      map[string]string `json:"info"`
```

Comment:
I'm not sure that declaring subtypes for Scheduler and Worker adds anything. 

Something like this might make more sense:
```golang
// DaskClusterSpec defines the desired state of DaskCluster
type DaskClusterSpec struct {
	Scheduler corev1.PodTemplate `json:"scheduler"`
	Worker    corev1.PodTemplate `json:"worker"`
	Info      map[string]string  `json:"info"`
	Replicas  int32              `json:"replicas,omitempty"`
}
```

It's clear the replicas refers to workers, but we can always label it workerReplicas if necessary.

Reply:
Hm, on second thought, the type will probably need to be a byte slice. The controller is going to have to deserialize the contents into corev1.PodTemplates.

Reply:
I originally had that, but wanted to match other kubernetes objects that used pod templates (where there's a `template: ...` field the template goes under). Since there will be other things we'll need to add to the spec (ports exposed on the scheduler, etc...), I'm inclined to leave the nesting for now unless it proves problematic.

Reply:
Copy that. Makes sense!